.github/workflows/import_count.yml

```
@@ -0,0 +1,69 @@
+name: Compute transitive imports
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.8
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        sudo apt-get install -y jq
+        # If you have additional dependencies, install them here
+
+    - name: Get changed files
+      run: |
+        git fetch origin ${{ github.base_ref }}  # fetch the base branch
+        git diff --name-only origin/${{ github.base_ref }}... > changed_files.txt  # get the list of changed files
+        cat changed_files.txt
+
+    - name: Compute transitive imports
+      run: |
+        # the checkout dance, to avoid a detached head
+        git checkout master
+        git checkout -
+
+        # Compute the counts for the HEAD of the PR
+        python ./scripts/count-trans-deps.py "Mathlib/" > head.json
+
+        # Checkout the merge base
+        git checkout master # let's start easy! "$(git merge-base refs/remotes/origin/main ${{ github.sha }})"
+        # get the files that we need from the current PR
+        # this will have to switch to `master` probably, when merging or right after
+        git checkout origin/import-count scripts/count-trans-deps.py scripts/import-graph-report.py
+
+        # Compute the counts for the merge base
+        python ./scripts/count-trans-deps.py "Mathlib/" > base.json
+
+    - name: Compare results and comment
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        message=$(python ./scripts/import-graph-report.py base.json head.json changed_files.txt)
+        echo "${message}"
+        data=$(jq -n --arg msg "$message" '{"body": $msg}')
+        method="POST"
+        if [[ -n "$message" ]]; then
+            url="https://api.github.com/repos/${GITHUB_REPOSITORY}/issues/${{ github.event.pull_request.number }}/comments"
+            headers="Authorization: token ${GITHUB_TOKEN}"
+            comments=$(curl -s -S -H "Content-Type: application/json" -H "$headers" "$url")
+            comment_id=$(echo "$comments" | jq -r '.[] | select(.body | startswith("### Import summary")) | .id')
```

```suggestion
            comment_id=$(curl -s -S -H "Content-Type: application/json" -H "$headers" "$url" |
               jq -r '.[] | select(.body | startswith("### Import summary")) | .id')
```
Also, is `comment_id` actually a line-break-separated list of comment id's that fit the query, or is it just the first one?